catalogue/webapp/components/ConfirmItemRequest/ConfirmItemRequest.tsx

```diff
@@ -184,37 +202,91 @@ const ErrorDialog: FunctionComponent<ErrorDialogProps> = ({ setIsActive }) => (
   </>
 );
 
+type RequestingState = null | 'requesting' | 'confirmed' | 'error';
+
 const ConfirmItemRequest: FunctionComponent<Props> = props => {
   const openButtonRef = useRef<HTMLButtonElement>(null);
-  const { item, work, setIsActive, ...modalProps } = props;
-  const [isConfirmed, setIsConfirmed] = useState(false);
-  const [isLoading, setIsLoading] = useState(false);
-  const [isError, setIsError] = useState(false); // TODO: implement this if something goes wrong with API call
-
+  const { item, work, setIsActive, user, initialHoldNumber, ...modalProps } =
+    props;
+  const [requestingState, setRequestingState] = useState<RequestingState>(null);
+  const [currentHoldNumber, setCurrentHoldNumber] = useState(initialHoldNumber);
   function innerSetIsActive(value: boolean) {
     if (value) {
       setIsActive(true);
-    } else if (isLoading) {
+    } else if (requestingState === 'requesting') {
       // disable close dialog button during api call
```

Comment:
Is this a TODO?

Reply:
It's a comment David left explaining what's happening, but it's not that clear - I'll rewrite this function